src/qilib/utils/storage/mongo.py

```diff
@@ -154,46 +160,94 @@ def _retrieve_value_by_tag(self, tag: TagType, parent: ObjectId) -> Any:
             elif 'value' not in doc:
                 raise NoDataAtKeyError(f'Tag "{tag[0]}" is not a leaf')
             else:
-                return doc['value']
+                if field is None:
+                    return doc['value']
+                elif field in doc['value']:
+                    return doc['value'][field]
+                else:
+                    raise NoDataAtKeyError(f'The field "{field}" does not exists')
 
         else:
             doc = self._collection.find_one({'parent': parent, 'tag': tag[0], 'value': {'$exists': False}})
             if doc is None:
                 raise NoDataAtKeyError(f'Tag "{tag[0]}" cannot be found')
             else:
-                return self._retrieve_value_by_tag(tag[1:], doc['_id'])
+                return self._retrieve_value_by_tag(tag[1:], doc['_id'], field)
+
+    def _retrieve_individual_value_by_tag(self, field: Any, tag: TagType, parent: ObjectId) -> Any:
+        """Traverse the tree and retrieve the value of a field present in given leaf tag
 
-    def _store_value_by_tag(self, tag: TagType, data: Any, parent: ObjectId) -> None:
+        Args:
+            field: The name of the field to be retrieved
+            tag: The leaf tag
+            parent: The ObjectID of the leaf's parent
+
+        Returns:
+            The field Data held by the leaf
+
+        """
+        return self._retrieve_value_by_tag(tag, parent, field)
+
+    def _store_value_by_tag(self, tag: TagType, data: Any, parent: ObjectId, field: Any = None) -> None:
         """ Store a value at a given tag
 
         Args:
             tag: The tag
             data: Data to store
             parent: An ObjectID of the node's parent
+            field: Field to be updated. Default value is None
```

Comment:
Actually this routine is not suitable for storing field values (case field is not None and non-existent). Field values can only be updated. Because of the re-use it is now in here. Because it is not part of the public interface I think it is oke, but it should be clearly described in the docstring.